chore: Document messages routing behaviour for references (#476)
SAP/ai-sdk: docs-js/orchestration/chat-completion.mdx

});
```

:::info Messages with Prompt Template References
The `prompt` property is optional when storing a template in the Prompt Registry, meaning a template can be stored without defining any prompt text.
In that case, provide messages via the `messages` property in `chatCompletion()` or `stream()` — they will be routed to `messages_history` automatically.

> When using a `template_ref`, messages passed to `chatCompletion()` or `stream()` are automatically appended to `messages_history` rather than merged into the remote template.

Reviewer: [pp] I think whether it is optional is not relevant here.

```suggestion
When using a `template_ref`, messages passed to `chatCompletion()` or `stream()` are automatically appended to `messages_history` as they cannot be merged into the remote template.
```

Author: Should we mention it elsewhere?

Reviewer: It should be fine not to mention, as long as we don't assert it's not optional.

> The messages are appended after any existing `messagesHistory` entries.

Reviewer: [pp]
```suggestion
The messages defined in the template are appended after any existing `messagesHistory` entries.
```

> All configured orchestration modules are applied to `messages_history`.

Reviewer: [q] Not sure if we want to assert this.

```suggestion
```

Author: Maybe my wording was not correct here that they always apply, I just wanted to mention that the messages in messages_history are treated the same. But yeah I can remove it :)

Reviewer: It should mostly be the same, but e.g. they are excluded from prompt templating so not completely the same.

console.log(response.getContent());
```

> :::info Messages with Orchestration Configuration References

Reviewer: [pp] Please edit likewise.
